addons/binding/org.openhab.binding.regoheatpump/src/main/java/org/openhab/binding/regoheatpump/internal/rego6xx/AbstractResponseParser.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.regoheatpump.internal.rego6xx;
+
+import javax.xml.bind.DatatypeConverter;
+
+/**
+ * The {@link AbstractResponseParser} is responsible for parsing responses coming from
+ * rego6xx controllers.
+ *
+ * @author Boris Krivonog - Initial contribution
+ */
+abstract class AbstractResponseParser<T> implements ResponseParser<T> {
+    private final static byte ComputerAddress = (byte) 0x01;
+
+    @Override
+    public abstract int responseLength();
+
+    protected abstract T convert(byte[] responseBytes);
+
+    @Override
+    public T parse(byte[] buffer) {
+        if (buffer == null) {
+            throw new NullPointerException();
```

Comment:
Its not a good practice to throw the NPE yourself, you can use Objects.requireNonNull to help to fail quickly. Although if you leave this out, you will just get an NPE at line 33. So if its really not expected, just leave out the check.